### DISCUSSION TITLE
Collect bacnet data using bacpypes

### DISCUSSION THREAD

**Discussion by yuleib** [JoelBender/bacpypes#423] [Q&A]
Hi i'm new with bacnet and bacpypes. 

Trying to understand how can i collect data which arriving to bacnet server using bacpypes
(bacpypes code is located at the server)

Thank you :)

**Comment by JoelBender** [comment]
I completely missed this discussion question!

The simplest way is to look at the **MultipleReadProperty.py** where you provide a list of points to read and when it's done, zip up the data and send it someplace.  This particular sample runs once and exits, so you could run this an a "cron job" at some interval or you could modify it to loop around and read the points again.

It should be pretty simple to extract a point list from a database, when you get to multiple intervals it gets a little harder, when you want to take advantage of **Read-Property-Multiple** it gets interesting.  The **MultipleReadPropertyHammer.py** application sends multiple concurrent reads to the device at the same time which can quickly overwhelm a device or the network if it's a field bus like MS/TP or ARCNET.
